### PR TITLE
Support API v17 for SDP Operator

### PR DESCRIPTION
**Pull request by mandopaloooza**
Operator Changes: 
* Fix the discriminator parser so it can read the updated Identity Provider API spec
* Support v14 ~ v17 API spec 
* Add bunch of unit testing for API v14 ~ v17 
* Add version suffix to CRDs

Helm Chart Changes: 
* Decouple `.sdp.operator.image.tag` and `.sdp.operator.version` so that you can specify a customized image when testing the operator in k8s environment
* Convert 0/1 values into true/false for boolean values
* Add JSON schema validation for values.yaml 
* Update and reorganize values.yaml into required and optional values 
* Store the CRDs into charts. During build, make sure that the generated CRD and checked-in CRDs are the same

### CODE REVIEW

File: appgate/openapi/openapi.py [hunk 1]
```diff
@@ -209,6 +209,19 @@ def add_items_key_to_tags(obj: dict) -> dict:
 
     schema = add_items_key_to_tags(schema)
 
+    # replace underscore with period
+    def replace_underscore(obj: dict) -> dict:
```
Comment (thomascellerier) [inline appgate/openapi/openapi.py:213]: Just a nitpick but seems like the comment could a docstring instead, also mention its recursive.
```suggestion
    def replace_underscore(obj: dict) -> dict:
        """
        Replace underscore with period recursively.
        """
```

File: appgate/openapi/parser.py [hunk 2]
```diff
@@ -387,30 +418,42 @@ def parse_all_of(self, definitions: List[OpenApiDict]) -> OpenApiDict:
...
-                new_definition["discriminator"]["mapping"] = new_mapping
-                new_definition["properties"] = new_properties
-
+            new_definition["discriminator"].update(d.get("discriminator", {}))
             if "description" in d:
                 descriptions.append(d["description"])
         new_definition["description"] = ".".join(descriptions)
         return new_definition
 
+    def parse_discriminator(self, definition: Any) -> OpenApiDict:
```
Comment (thomascellerier) [inline appgate/openapi/parser.py:427]: A docstring would be nice to explain what kind of transformation this does.
Is definition really `Any`? It looks like its some form of mapping at least.

**Review by thomascellerier — APPROVED** [review]
Looks good.

File: appgate/openapi/parser.py (hunk 2 above)
Comment (mandopaloooza) [reply appgate/openapi/parser.py:427]: Good catch. Should be `OpenApiDict` type.

**Pull request merged**